redfish/provider/data_source_redfish_storage.go

```diff
@@ -142,9 +148,635 @@ func (g *StorageDatasource) readDatasourceRedfishStorage(d models.StorageDatasou
+		RealtimeCapability:               types.StringValue(input.RealtimeCapability),
+		RollupStatus:                     types.StringValue(input.RollupStatus),
+		SASAddress:                       types.StringValue(input.SASAddress),
+		SecurityStatus:                   types.StringValue(input.SecurityStatus),
+		SharedSlotAssignmentAllowed:      types.StringValue(input.SharedSlotAssignmentAllowed),
+		SlicedVDCapability:               types.StringValue(input.SlicedVDCapability),
+		SupportControllerBootMode:        types.StringValue(input.SupportControllerBootMode),
+		SupportEnhancedAutoForeignImport: types.StringValue(input.SupportEnhancedAutoForeignImport),
+		SupportRAID10UnevenSpans:         types.StringValue(input.SupportRAID10UnevenSpans),
+		SupportsLKMtoSEKMTransition:      types.StringValue(input.SupportsLKMtoSEKMTransition),
+		T10PICapability:                  types.StringValue(input.T10PICapability),
+	}
+}
+
+// newDellControllerBattery converts redfish.DellControllerBattery to models.DellControllerBattery
+func newDellControllerBattery(input dell.DellControllerBattery) models.DellControllerBattery {
+	return models.DellControllerBattery{
+		OdataContext:  types.StringValue(input.OdataContext),
+		OdataID:       types.StringValue(input.OdataID),
+		OdataType:     types.StringValue(input.OdataType),
+		Description:   types.StringValue(input.Description),
+		Fqdd:          types.StringValue(input.Fqdd),
+		ID:            types.StringValue(input.ID),
+		Name:          types.StringValue(input.Name),
+		PrimaryStatus: types.StringValue(input.PrimaryStatus),
+		RAIDState:     types.StringValue(input.RAIDState),
+	}
+}
+
+// newDell converts redfish.Dell to models.Dell
+func newDell(input dell.StorageOEM) models.Dell {
+	return models.Dell{
+		OdataType:             types.StringValue(input.OdataType),
+		DellController:        newDellController(input.DellController),
+		DellControllerBattery: newDellControllerBattery(input.DellControllerBattery),
+	}
+}
+
+// newOem converts redfish.Oem to models.Oem
+func newOem(input dell.StorageOEM) models.Oem {
+	return models.Oem{
+		Dell: newDell(input),
+	}
+}
+
+// newStatus converts redfish.Status to models.Status
+func newStatus(input common.Status) models.Status {
+	return models.Status{
+		Health:       types.StringValue(string(input.Health)),
+		HealthRollup: types.StringValue(string(input.HealthRollup)),
+		State:        types.StringValue(string(input.State)),
+	}
+}
+
+// newCacheSummary converts redfish.CacheSummary to models.CacheSummary
+func newCacheSummary(input redfish.CacheSummary) models.CacheSummary {
+	return models.CacheSummary{
+		TotalCacheSizeMiB: types.Int64Value(int64(input.TotalCacheSizeMiB)),
+	}
+}
+
+// StorageSchema is a function that returns the schema for Storage
+func StorageSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"odata_context": schema.StringAttribute{
```

Comment:
We generally do not expose odata fields, like odata_context,  odata_type, or *_odata_count from our providers. Please confirm if these fields are expected here.

Reply:
removed all the odata fields